src/main/java/webdriver/WebDriverFactory.java

```diff
@@ -13,35 +14,22 @@
 
     protected abstract WebDriver createDriver();
 
-    public static WebDriverFactory getDriverFactory() {
-        return getPlaceOfLaunch();
-    }
-
-    private static WebDriverFactory getPlaceOfLaunch() {
+    public static WebDriverFactory getDriverFactory(String browser) {
         String placeLaunch = System.getProperty("placeLaunch", getBrowserProperty("placeLaunch"));
 
         if (equalsIgnoreCase(placeLaunch, "localLaunch")) {
-            return getLocalLaunch();
+            if (equalsIgnoreCase(browser, CHROME)) {
+                return new ChromeDriverFactory();
+            } else if (equalsIgnoreCase(browser, FIREFOX)) {
+                return new FirefoxDriverFactory();
+            } else {
+                throw new RuntimeException(String.format("%s are not supported.", browser));
+            }
         } else if (equalsIgnoreCase(placeLaunch, "remoteLaunch")) {
-            return getRemoteLaunch();
+            return new RemoteDriverFactory(browser);
         } else {
             throw new RuntimeException();
```

Comment:
Best practice is to add message if you throw common Exception. Or even better to create your own, like
DriverFactoryInitializationException which points to specific issue comparing to RuntimeException